fenix/app/src/main/java/org/mozilla/fenix/shopping/state/ReviewQualityCheckPreferencesMiddleware.kt

```diff
@@ -0,0 +1,91 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.fenix.shopping.state
+
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.launch
+import mozilla.components.lib.state.Middleware
+import mozilla.components.lib.state.MiddlewareContext
+import mozilla.components.lib.state.Store
+
+/**
+ * Middleware for getting and setting review quality check user preferences.
+ *
+ * @param reviewQualityCheckPreferences The [ReviewQualityCheckPreferences] instance to use.
+ * @param scope The [CoroutineScope] to use for launching coroutines.
+ */
+class ReviewQualityCheckPreferencesMiddleware(
+    private val reviewQualityCheckPreferences: ReviewQualityCheckPreferences,
+    private val scope: CoroutineScope,
```

Comment:
I think we could create a default parameter for the scope here so that it didn't need to be propagated through the type hierarchy

Reply:
That would not auto cancel the coroutines when the fragment is destroyed, the idea is to use [lifecycleScope](https://developer.android.com/topic/libraries/architecture/coroutines#lifecyclescope) so any long running tasks are auto cancelled when the  fragment is destroyed.